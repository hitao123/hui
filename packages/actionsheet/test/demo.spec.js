import { shallowMount } from '@vue/test-utils';
import { renderToString } from '@vue/server-test-utils';
import Actionsheet from '../index.vue';
import ActionsheetDemo from '../demo/index.vue';

describe('Actionsheet component', () => {
  it('should show Actionsheet', () => {
    const warpper = shallowMount(Actionsheet, {
      propsData: {
        title: 'actionsheet',
        actions: [
          {
            name: '打开相机'
          },
          {
            name: '选择照片',
            subname: 'tips'
          },
          {
            loading: true
          }
        ]
      }
    });
    expect(warpper.html()).toMatchSnapshot();
  });

  it('should show Actionsheet Demo', () => {
    expect(renderToString(ActionsheetDemo)).toMatchSnapshot();
  });
});
