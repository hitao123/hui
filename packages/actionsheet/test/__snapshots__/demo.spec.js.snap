// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actionsheet component should show Actionsheet 1`] = `
<div class="h-actionsheet h-actionsheet--withtitle" style="display: none;" name="slide-bottom">
  <div class="h-hairline--top-bottom h-actionsheet__header">
    <div>actionsheet</div>
    <icon-stub name="close" classprefix="h-icon"></icon-stub>
  </div>
  <div class="h-actionsheet__content"></div>
</div>
`;

exports[`Actionsheet component should show Actionsheet Demo 1`] = `
<demo-section>
  <demo-block title="基本用法">
    <h-button>action title</h-button>
  </demo-block>
  <demo-block title="待取消的 ActionSheet">
    <h-button>action 列表</h-button>
  </demo-block>
  <demo-block title="带title的 ActionSheet">
    <h-button>action 取消</h-button>
  </demo-block>
  <h-actionsheet title="title">
    <p>内容</p>
  </h-actionsheet>
  <h-actionsheet actions="[object Object],[object Object],[object Object]"></h-actionsheet>
  <h-actionsheet title="title" cancelText="取消"></h-actionsheet>
</demo-section>
`;
